glean/src/core/metrics/types/url.ts

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { CommonMetricData } from "../index.js";
+import { isString } from "../../utils.js";
+import { MetricType } from "../index.js";
+import { Context } from "../../context.js";
+import { Metric } from "../metric.js";
+import { ErrorType } from "../../error/error_type.js";
+
+// The maximum number of characters a URL Metric may have, before encoding.
+const URL_MAX_LENGTH = 2048;
+
+// This regex only validates that the `scheme` part of the URL is spec compliant
+// and is followed by a ":". After that anything is accepted.
+//
+// Reference: https://url.spec.whatwg.org/#url-scheme-string.
```

Comment:
```suggestion
// Reference: https://url.spec.whatwg.org/#url-scheme-string
```

just to make the url easily clickable (`.` is a valid character, so most parser will otherwise include it, breaking the anchor)